crates/oxc_semantic/src/symbol.rs

```diff
@@ -62,10 +62,13 @@ impl SymbolTable {
         self.spans.iter_enumerated().map(|(symbol_id, _)| symbol_id)
     }
 
+    // Symbol are created in dfs order, so we make two assumptions:
```

Comment:
Let's not make this assumption.

This API is added for the import plugin, are you using this anywhere else?

Reply:
Yes. That assumption will not hold if the AST has been through transformer. Right now I believe Rolldown re-runs semantic after transformer, but we are working to keep `Semantic` in sync with AST in transformer, to make that unnecessary (which will be a decent speed-up).